app/src/main/java/ar/com/wolox/android/cookbook/mpchart/MpChartRecipeFragment.kt

```diff
@@ -0,0 +1,357 @@
+package ar.com.wolox.android.cookbook.mpchart
+
+import android.app.ProgressDialog
+import android.view.View
+import android.widget.AdapterView
+import android.widget.Toast
+import ar.com.wolox.android.cookbook.R
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment
+import com.github.mikephil.charting.components.XAxis
+import com.github.mikephil.charting.data.BarData
+import com.github.mikephil.charting.data.BarEntry
+import com.github.mikephil.charting.data.BubbleData
+import com.github.mikephil.charting.data.BubbleEntry
+import com.github.mikephil.charting.data.CandleData
+import com.github.mikephil.charting.data.CombinedData
+import com.github.mikephil.charting.data.Entry
+import com.github.mikephil.charting.data.LineData
+import com.github.mikephil.charting.data.PieData
+import com.github.mikephil.charting.data.PieEntry
+import com.github.mikephil.charting.data.RadarData
+import com.github.mikephil.charting.data.ScatterData
+import com.github.mikephil.charting.highlight.Highlight
+import com.github.mikephil.charting.listener.OnChartValueSelectedListener
+import kotlinx.android.synthetic.main.fragment_mp_chart.*
+
+/**
+ * Note: Many of the chart have the same parameters of configuration, specific parameters are defined
+ * in each method.
+ *
+ * Parameters:
+ * data (mandatory): data to display in chart
+ *
+ * animations -> Animates the rendering of the chart on the x/y-axis with the specified animation time
+ * animateX
+ * animateY
+ * animateXY
+ *
+ * isDragEnabled: Enables dragging (moving chart with the finger) for chart bigger than the screen
+ *
+ * setOnChartValueSelectedListener(): set a listener for the chart, this returns an Entry object
+ * (Warning: in combined chart all types of charts return the same Entry type on item click)
+ *
+ * xAxis/yAxis: set position of labels, margins, etc...
+ *
+ * axisRight/axisLeft: Returns the _-axis object
+ *
+ * description.text: text with a short description of the chart (in the right bottom of the screen)
+ */
+class MpChartRecipeFragment : WolmoFragment<MpChartRecipePresenter>(), MpChartRecipeView {
+
+    private lateinit var progressDialog: ProgressDialog
+
+    override fun layout(): Int = R.layout.fragment_mp_chart
+
+    override fun init() {
+        progressDialog = ProgressDialog(context)
+        presenter.onInit()
```

Comment:
We can avoid this, and in presenter do directly onViewAttached on presenter. There you can init what it's necessary to initialize